src/net.c

```diff
@@ -303,7 +303,9 @@ static void periodic_handler(void *data) {
 
 void handle_meta_connection_data(connection_t *c) {
 	if(!receive_meta(c)) {
-		if(!c->status.control) {
+		// Terminate invitation connections right away since
+		// hanging connections confuse and frustrate users
+		if(!c->status.control && !c->status.invitation) {
```

Comment:
I did say we should not spend resources on botched join connections, but we should tarpit those connections anyway. The tarpit will automatically close connections if there are too many of them, so it only uses a fixed number of filedescriptors. We do want to frustrate attackers who are trying joins.